builds/jazz-build-module/aws-lambda-events-module.groovy

```diff
@@ -0,0 +1,194 @@
+  def events = action.split(",")
+  def lambdaFunctionConfigurations = []
+  def new_events = []
+  new_lambda_configuration.LambdaFunctionArn = lambdaARN
+
+  try {
+    if (existing_notifications.containsKey("LambdaFunctionConfigurations")) {
+      def existing_lambda_configs = existing_notifications.LambdaFunctionConfigurations
+      new_events = getLambdaEvents(existing_notifications, events)
+      if (new_events != null && new_events.size() > 0) {
+        for (item in existing_lambda_configs) {
+          lambdaFunctionConfigurations.add(item)
+        }
+        new_lambda_configuration.Events = new_events
+        lambdaFunctionConfigurations.add(new_lambda_configuration)
+      }
+    } else {
+      new_events = checkAndConvertEvents(events)
+      new_lambda_configuration.Events = new_events
+      lambdaFunctionConfigurations.add(new_lambda_configuration)
+    }
+    if (lambdaFunctionConfigurations != null && lambdaFunctionConfigurations.size() > 0) {
+      existing_notifications.LambdaFunctionConfigurations = lambdaFunctionConfigurations
+      def newNotificationJson = JsonOutput.toJson(existing_notifications)
+      def response = sh(returnStdout: true, script: "aws s3api put-bucket-notification-configuration --bucket $s3BucketName --notification-configuration \'${newNotificationJson}\' --output json")
+    }
+  } catch (ex) {
+  }
+}
+
+def getLambdaEvents(existing_notifications, events){
+  def existing_events = []
+  def new_events = []
+  for (item in events) {
+    new_events.add(item)
+  }
+  for (item in existing_notifications.LambdaFunctionConfigurations) {
+    existing_events.addAll(item.Events)
+  }
+  def cleanupIndex = -1
+  echo "events . $events"
+
+  // Removing the existing events from the new event list
+  for (item in events) {
+    cleanupIndex++
+    if ((item.contains("ObjectCreated") || item.contains("ObjectRemoved")) &&
+      (existing_events.contains("s3:ObjectCreated:*") || existing_events.contains("s3:ObjectRemoved:*"))) {
+      new_events[cleanupIndex] = null
+    } else if (existing_events.contains(item)) {
+      new_events[cleanupIndex] = null
+    }
+  }
+  new_events.removeAll([null])
+  def events_list = []
+  if (new_events.size() > 0 && new_events != null) {
+    events_list = checkAndConvertEvents(new_events)
+  }
+  return events_list
+}
+
+def checkAndConvertEvents(events){
+  def new_events = []
+  // converting the new events to * event
+  def cleanupIndex = -1
+  def isCreationEvent = false
+  def isRemovalEvent = false
+
+  if (events.size() > 0 && events != null) {
+    for (item in events) {
+      new_events.add(item)
+    }
+
+    if(new_events.contains("s3:ObjectRemoved:*")){
+      isRemovalEvent = true
+    }
+
+    for (item in events) {
+      cleanupIndex++
+      if (item.contains("ObjectCreated")) {
+        isCreationEvent = true
+        new_events[cleanupIndex] = null
+      }
+      if(isRemovalEvent == true && item.contains("ObjectRemoved")){
```

Comment:
I had adjusted with tabs and spaces. And still it is showing correct in IDE but not over here. Since it is taking more time for testing after each fix, even though  these are minor issues. It will be better if some one can provide any plugin for jenkins file indentaion fixing as like editorconfig for javascript.